### ISSUE TITLE
Notifications: improve Blockquote block handling

### ISSUE THREAD

**Issue by lancewillett** [Automattic/wp-calypso#41331]
#### Steps to reproduce

First, on any WordPress.com site, post a comment with a blockquote in it.

<img width="373" alt="Screen Shot 2020-04-21 at 09 29 50" src="https://user-images.githubusercontent.com/66797/79889286-a8585500-83b2-11ea-8e66-3625f78950bf.png">

Then, make a comment on a P2-themed WordPress.com site that uses Gutenberg block editor.

<img width="843" alt="Screen Shot 2020-04-21 at 09 02 41" src="https://user-images.githubusercontent.com/66797/79887308-637eef00-83af-11ea-8a27-cd13efc0d634.png">

#### What I expected

To see the same behavior and visual styling in both cases.

#### What happened instead

The normal comment form input looks as expected. The quote is in italics and indented a bit.

<img width="413" alt="Screen Shot 2020-04-21 at 09 01 40" src="https://user-images.githubusercontent.com/66797/79887209-35011400-83af-11ea-82c5-3ac4873584b2.png">

Here's the comment using Gutenberg blocks. Note: missing the blockquote styling.

<img width="413" alt="Screen Shot 2020-04-21 at 09 03 48" src="https://user-images.githubusercontent.com/66797/79887319-6a0d6680-83af-11ea-8682-1eba30f4ed86.png">

#### Context / Source

Reported in normal use by @m

**Comment by dmsnell** [comment]
@lancewillett this doesn't look easy. we have an odd interaction between the `the_comment` filter, the `do_blocks()` processing, and the client code. the API is returning the right information but the notifications panel in Calypso and in the masterbar is interpreting it wrong, largely because of extra newlines.

I'm not sure what timeline I can give on fixing this but if anyone can help I can definitely guide them in. we're probably looking at `indices-to-html` as the culprit. I tried a few simple fixes on the server and nothing worked without causing worse side-effects.

**Comment by lancewillett** [comment]
Thanks for digging into it and sharing notes, Dennis.

Here is what I saw in the HTML source, for reference.

This is the "plain text" comment created in the notifications panel:

```
<blockquote>
<p>I can see! Said the blind carpenter, as they picked up their hammer and saw.</p>
</blockquote>
```

This is the markup generated by the Gutenberg block comment via P2tenberg:

```
<blockquote class="wp-block-quote"><p>This is a Gutenberg <code>Quote</code> block added in a comment on front-end of P2 with P2tenberg editor.</p></blockquote>
```

**Comment by dmsnell** [comment]
That's correct @lancewillett, and the notifications API properly indicates that the stretch of text is a blockquote. What happens with Blocks though is that outputs something like this…

```js
{
	text: "\n\nThis is a Gutenberg Quote block added in a comment.\n",
	ranges: [
		{ type: 'blockquote', indices: [ 0, 53 ] }
	]
}
```

For some reason the notes panel creates an _empty_ blockquote and then a new paragraph after it, which you can see if you inspect the generated HTML inside the notifications panel. It's splitting the message on that double-newline or something like that. The mobile apps appear to handle this properly and I suspect a problem with the parsing code which has never been touched since it was written - `indices-to-html` in the code.

**Comment by lancewillett** [comment]
Re-testing in 2025 ("Back to the future") style.

This is working now! Hurray.

Made a new comment using block editor.

![Image](https://github.com/user-attachments/assets/d2080aed-3d67-4a13-9c4b-8fb5094a24a0)

Here is the notification in all its visual glory:

![Image](https://github.com/user-attachments/assets/385afcbe-8ebf-44a0-ab0f-51e7a11c0398)
